File: php/tests/Fixtures/InMemoryRepository.php
<?php

namespace Tests\Katalogs\BookInvoicing\Fixtures;

use Katalogs\BookInvoicing\Domain\Purchase\Invoice;
use Katalogs\BookInvoicing\Domain\Purchase\Repository;

class InMemoryRepository implements Repository
{
    private array $invoices = [];

    public function __construct()
    {
    }

    public function addInvoice(Invoice $invoice): void
    {
        $this->invoices[] = $invoice;
    }

    public function getInvoices(): array
    {
        return $this->invoices;
    }
}

File: python/tests/in_memory_repository.py
from typing import Dict
from src.book_invoicing.domain.purchase import Invoice
from src.book_invoicing.domain.repository import Repository


class InMemoryRepository(Repository):
    def __init__(self):
        self._invoice_map: Dict[int, Invoice] = {}

    def add_invoice(self, invoice: Invoice):
        self._invoice_map[invoice.id] = invoice

    def get_invoice_map(self) -> Dict[int, Invoice]:
        return self._invoice_map
